src/ui/DynamicHierarchicalFacetSearch/DynamicHierarchicalFacetSearchValueRenderer.ts

```diff
@@ -0,0 +1,118 @@
+import { IDynamicHierarchicalFacet } from '../DynamicHierarchicalFacet/IDynamicHierarchicalFacet';
+import { DynamicHierarchicalFacetSearchValue } from './DynamicHierarchicalFacetSearchValue';
+import { $$ } from '../../utils/Dom';
+import { l } from '../../strings/Strings';
+
+const PATH_START_LENGTH = 1;
+const PATH_END_LENGTH = 2;
+const PATH_SEPARATOR = '/';
+
+const VALUE_CLASSNAME = 'coveo-dynamic-hierarchical-facet-search-value';
+const HEADER_CLASSNAME = `${VALUE_CLASSNAME}-header`;
+const LABEL_CLASSNAME = `${VALUE_CLASSNAME}-label`;
+const COUNT_CLASSNAME = `${VALUE_CLASSNAME}-results-count`;
+const PATH_CLASSNAME = `${VALUE_CLASSNAME}-path`;
+const PATH_ELLIPSIS_CLASSNAME = `${PATH_CLASSNAME}-ellipsis`;
+const PATH_PART_CLASSNAME = `${PATH_CLASSNAME}-part`;
+const PATH_SEPARATOR_CLASSNAME = `${PATH_CLASSNAME}-separator`;
+
+export const DynamicHierarchicalFacetSearchValueRendererClassNames = {
+  VALUE_CLASSNAME,
+  HEADER_CLASSNAME,
+  LABEL_CLASSNAME,
+  COUNT_CLASSNAME,
+  PATH_CLASSNAME,
+  PATH_ELLIPSIS_CLASSNAME,
+  PATH_PART_CLASSNAME,
+  PATH_SEPARATOR_CLASSNAME
+};
+
+export class DynamicHierarchicalFacetSearchValueRenderer {
+  constructor(private facetValue: DynamicHierarchicalFacetSearchValue, private facet: IDynamicHierarchicalFacet) {}
+
+  private get pathToRender(): { start: string[]; end?: string[] } {
+    const parentPath = this.facetValue.fullPath.slice(0, -1);
+    if (!parentPath.length) {
+      return { start: [this.facet.options.clearLabel] };
+    }
+    if (parentPath.length > PATH_START_LENGTH + PATH_END_LENGTH) {
+      return { start: parentPath.slice(0, PATH_START_LENGTH), end: parentPath.slice(-PATH_END_LENGTH) };
+    }
+    return { start: parentPath };
+  }
+
+  private get label() {
+    const { start, end } = this.pathToRender;
+    return l(
+      'HierarchicalFacetValueIndentedUnder',
+      l(
+        'SelectValueWithResultCount',
+        this.facetValue.displayValue,
+        l('ResultCount', this.facetValue.numberOfResults, this.facetValue.numberOfResults)
+      ),
+      [...start, ...(end || [])].reverse().join(', ')
+    );
+  }
+
+  public render() {
```

Comment:
Overall, a render should look like:
```html
<div class="coveo-dynamic-hierarchical-facet-search-value" aria-label="Select John Doe with 1337 results under People, Canada, Milky Way">
  <header class="coveo-dynamic-hierarchical-facet-search-value-header" aria-hidden="true">
    <span class="coveo-dynamic-hierarchical-facet-search-value-label">John Doe</span>
    <span class="coveo-dynamic-hierarchical-facet-search-value-results-count">(1337)</span>
  </header>
  <ul class="coveo-dynamic-hierarchical-facet-search-value-path" aria-hidden="true">
    <li class="coveo-dynamic-hierarchical-facet-search-value-path-part">Milky Way</li>
    <li class="coveo-dynamic-hierarchical-facet-search-value-path-separator">/</li>
    <li class="coveo-dynamic-hierarchical-facet-search-value-path-ellipsis">...</li>
    <li class="coveo-dynamic-hierarchical-facet-search-value-path-separator">/</li>
    <li class="coveo-dynamic-hierarchical-facet-search-value-path-part">Canada</li>
    <li class="coveo-dynamic-hierarchical-facet-search-value-path-separator">/</li>
    <li class="coveo-dynamic-hierarchical-facet-search-value-path-part">People</li>
  </ul>
</div>
```

Reply:
The classnames are pretty long.

But that's good, that means there will be no nesting needed to customize them :)

Reply:
Yeah, I felt like they were too long but I also felt like shortening them would make them too ambiguous.